simulations/vip-160/vip-160/simulation.ts

```diff
@@ -0,0 +1,166 @@
+import { expect } from "chai";
+import { Contract } from "ethers";
+import { parseUnits } from "ethers/lib/utils";
+import { ethers } from "hardhat";
+
+import { forking, testVip } from "../../../src/vip-framework";
+import { vip160 } from "../../../vips/vip-160/vip-160";
+import COMPTROLLER_ABI from "./abi/comptroller.json";
+import POOL_REGISTRY_ABI from "./abi/poolRegistry.json";
+import PROTOCOL_SHARE_RESERVE_ABI from "./abi/protocolShareReserve.json";
+import RISK_FUND_ABI from "./abi/riskFund.json";
+import SHORTFALL_ABI from "./abi/shortfall.json";
+import VTOKEN_ABI from "./abi/vToken.json";
+
+const NORMAL_TIMELOCK = "0x939bD8d64c0A9583A7Dcea9933f7b21697ab6396";
+const POOL_REGISTRY = "0x9F7b01A536aFA00EF10310A162877fd792cD0666";
+const USDT = "0x55d398326f99059fF775485246999027B3197955";
+const SWAP_ROUTER_CORE_POOL = "0x8938E6dA30b59c1E27d5f70a94688A89F7c815a4";
+const VENUS_TREASURY = "0xF322942f644A996A617BD29c16bd7d231d9F35E9";
+
+const SHORTFALL = "0x51dcdD4Ee45c0e8c04DE4C54908C2Ada38fd99B5";
+const PROTOCOL_SHARE_RESERVE = "0x99731dDeA6e0d721dfeb434Cb20b200E72709056";
+const RISK_FUND = "0xDF829fB0861DA07a0a935486738dFa65ED007927";
+
+interface VenusPool {
+  comptroller: string;
+}
+
+const getAllMarkets = async (poolRegistry: Contract) => {
+  const pools = await poolRegistry.getAllPools();
+  const comptrollers = await Promise.all(
+    pools.map((pool: VenusPool) => ethers.getContractAt(COMPTROLLER_ABI, pool.comptroller)),
+  );
+  const marketAddresses = (
+    await Promise.all(comptrollers.map(async (comptroller: Contract) => comptroller.getAllMarkets()))
+  ).flat();
+  const markets = await Promise.all(
+    marketAddresses.map((address: string) => ethers.getContractAt(VTOKEN_ABI, address)),
+  );
+  return markets;
+};
+
+forking(30903270, () => {
+  let poolRegistry: Contract;
+
+  testVip("Risk fund, shortfall, PSR, stage 1", vip160(), {
```

Comment:
We could verify the emitted events

Reply:
[c038a61](https://github.com/VenusProtocol/vips/pull/59/commits/c038a61b682c6cc16b1a45aa90f76eb19f2bf6a4)